cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +847,34 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::Balance() {
+  CatalogList catalog_list = GetCatalogs();
```

Comment:
Ah, in that case you are totally right. It should be locked.
However it's important to notice that the whole catalog manager would be locked for some seconds... or even minutes